### PR TITLE
python: passing arguments from `syslog-ng.conf`

### LINKED ISSUE #535: python: passing arguments from config to init()
It would be nice to parse arguments from syslog-ng.conf and pass it to '__init__()'  method in python destinations.

example:

```
@version: 3.7

source example_source {
         network(ip(127.0.0.1) port(1999));
};

destination python_http {
            python(
                class("http.HTTPDestination")
                arguments(server="localhost" port="80")
                on-error("fallback-to-string")
                value-pairs(scope(everything)));
};

log {
    source(example_source);
    destination(python_http);
};

```
```python
  def __init__(self, args):
      self.server_address = args['server_address']
```

### PR DESCRIPTION
**Pull request by gregory094**
Arguments can be passed to python destinations
using `options` token in the config file.

Example:
```
log {
  source { tcp(port(12345));  };
  destination {
  python(
    class("sngexample.DummyPythonDest")
    options(
      key1("value for key1")
      key2("value for key2")
    )
  );
  };
};
```

Fixes #535

Created-with: Arsenault Adam <adam.arsenault@balabit.com>
Signed-off-by: Gergő Nagy <gergo.nagy@balabit.com>

### CODE REVIEW

**Comment by gregory094** [comment]
@bazsi could u look at this PR if u have some time?

**Comment by lbudai** [comment]
travis says:
```
modules/python/python-dest.c: In function ‘_py_invoke_bool_method_by_name’:
modules/python/python-dest.c:255:3: warning: implicit declaration of function ‘_py_invoke_bool_method_by_name_with_args’ [-Wimplicit-function-declaration]
modules/python/python-dest.c: At top level:
modules/python/python-dest.c:259:1: error: static declaration of ‘_py_invoke_bool_method_by_name_with_args’ follows non-static declaration
modules/python/python-dest.c:255:10: note: previous implicit declaration of ‘_py_invoke_bool_method_by_name_with_args’ was here
modules/python/python-dest.c: In function ‘_py_invoke_init’:
modules/python/python-dest.c:302:3: warning: passing argument 1 of ‘python_dd_create_arg_dict’ from incompatible pointer type [enabled by default]
modules/python/python-dest.c:122:1: note: expected ‘struct LogDriver *’ but argument is of type ‘struct PythonDestDriver *’
make[2]: *** [modules/python/modules_python_libmod_python_la-python-dest.lo] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

File: modules/python/python-dest.c [hunk 1]
```diff
@@ -56,6 +57,23 @@ typedef struct
...
+  gchar *p;
+  gchar *normalized_key = g_strdup(buffer);
+  p = normalized_key;
+  while (*p)
+    {
+      if (*p == from)
+        *p = to;
+      p++;
+    }
+  return normalized_key;
```
Comment (lbudai) [inline modules/python/python-dest.c]: I would factor out a `str_replace_char(gchar *str, gchar from, gchar to)` function ( into `misc.c`).

Comment (lbudai) [inline modules/python/python-dest.c]: I think this `normalize_key` could be used also in java... but wait... : we also have this function in `java-destination.c`. Please move this function(with `str_replace_char`) into `misc.h/c` and remove from java and python.

**Comment by lbudai** [comment]
any update on this?

**Comment by lbudai** [comment]
hi,

I'll need your patches soon.
If you don't have the time, I can branch your work and can do the suggested modifications myself then you can update this PR. But without those changes I won't merge it. Code duplication is not a friend of mine :-(

**Comment by gregory094** [comment]
Sry... Code duplication is neither a friend of mine. :) I thought that I'd pushed my changes before. Here they are.

**Comment by gregory094** [comment]
This patch is corrected. Python-C reference counting fixed and tested.

**Comment by ihrwein** [comment]
Please resolve the conflicts and update the text of your PR with the new configuration syntax!.

**Comment by gregory094** [comment]
@lbudai could u look at this refactored PR?

**Comment by lbudai** [comment]
Looks good for me.

**Pull request merged**